beacon-chain/rpc/beaconv1/state_test.go

```diff
@@ -76,95 +76,90 @@ func TestGetGenesis(t *testing.T) {
 }
 
 func TestGetStateRoot(t *testing.T) {
+	db := testDB.SetupDB(t)
 	ctx := context.Background()
-	root := []byte("123456")
 
 	t.Run("Head", func(t *testing.T) {
-		chainService := &chainMock.ChainService{
-			Root: root,
-		}
+		b := testutil.NewBeaconBlock()
+		b.Block.StateRoot = bytesutil.PadTo([]byte("head"), 32)
 		s := Server{
-			ChainInfoFetcher: chainService,
+			ChainInfoFetcher: &chainMock.ChainService{Block: b},
 		}
+
 		resp, err := s.GetStateRoot(ctx, &ethpb.StateRequest{
 			StateId: []byte("head"),
 		})
 		require.NoError(t, err)
-		assert.DeepEqual(t, root, resp.Data.StateRoot)
+		assert.DeepEqual(t, resp.Data.StateRoot, resp.Data.StateRoot)
```

Comment:
You are comparing the same things here.
```suggestion
		assert.DeepEqual(t, bytesutil.PadTo([]byte("head"), 32), resp.Data.StateRoot)
```